### DISCUSSION TITLE
How to fix ERR_PNPM_LOCKFILE_MISSING_DEPENDENCY error when deploying a Firebase functions?

### DISCUSSION THREAD

**Discussion by JumBay** [pnpm/pnpm#6691] [Q&A]
Hi,

I have a monorepo project using NX.
I'm using pnpm@8.6.3 

When I try to deploy a Firebase functions, I get this error: 

> Build failed: installing pnpm dependencies: %!w(*buildererror.Error=&{  2 2 8622936f Lockfile is up to date, resolution step is skipped
>  ERR_PNPM_LOCKFILE_MISSING_DEPENDENCY  Broken lockfile: no entry for '/string-width/4.2.3' in pnpm-lock.yaml
> 
> This issue is probably caused by a badly resolved merge conflict.
> To fix the lockfile, run 'pnpm install --no-frozen-lockfile'.}); Error ID: d1512fdc

NX generates both `package.json` and `pnpm-lock.yaml` files, maybe the lock file is not valid. I can share them if it helps.

I don't know how to run `pnpm install --no-frozen-lockfile`. I deploy using the firebase cli: `firebase deploy --only functions:myFunction`

Thanks.

**Comment by plafreniere1** [comment]
I get exactly the same issue when deploying Firebase functions. Odly, both our package.json and pnpm-lock file have not changed since our last deploy and string-width 4.2.3 entry is there in lockfile... This error seems to just appeared out of the blue.

Nx monorepo, pnpm v.8.4.0

**Comment by simonmallet** [comment]
Same problem! I tried deleting my pnpm-lock.json file and running pnpm install, and it generates the exact same pnpm-lock.json file so nothing different in my lock file ; yet still get the same error

**Comment by seb-mynotary** [comment]
Same problème here.  Using nx monorepo too

**Comment by cjmzone** [comment]
Same problem here. I've tried deleting my pnpm-lock.json and running `pnpm install --no-frozen-lockfile` but no new lockfile is generated. This issue has been driving me crazy for over a week now.
